mergify_engine/redis_utils.py

```diff
@@ -231,6 +250,12 @@ def redis_from_url(
         url: str,
         decode_responses: bool,
         max_connections: typing.Optional[int] = None,
+        redis_connect_func: typing.Optional[
```

Comment:
Maybe you can create a custom TypeVar to only have to write that once ?

Reply:
Indeed good idea, too many repetitions !